### DISCUSSION TITLE
Statuses don't include GitHub Action

### DISCUSSION THREAD

**Discussion by sigfrid** [community/community#26839] [Actions]
<p>Hi there,<br>
In a repo, I have the following 4 checks</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7de1b0aadfbeb6db0e24bc323a28e87c442b671b.png" data-download-href="/uploads/short-url/hXBdNNcTQ4IOzwstiILscsLjWhZ.png?dl=1" title="Screen Shot 2020-09-23 at 11.40.51 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/3609259/181100249-b19586f7-d2ab-4fc8-9b84-8630c8435a4d.png" alt="Screen Shot 2020-09-23 at 11.40.51 AM" data-base62-sha1="hXBdNNcTQ4IOzwstiILscsLjWhZ" width="690" height="198" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7de1b0aadfbeb6db0e24bc323a28e87c442b671b_2_690x198.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7de1b0aadfbeb6db0e24bc323a28e87c442b671b_2_1035x297.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7de1b0aadfbeb6db0e24bc323a28e87c442b671b_2_1380x396.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7de1b0aadfbeb6db0e24bc323a28e87c442b671b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-09-23 at 11.40.51 AM</span><span class="informations">1686×484 61 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>However, if I try to their status via the API I get only those generated by codeclimate but not by the GitHub Action that runs CI. See below</p>
<p><a href="https://api.github.com/repos/:ORG/:REPO/commits/2961d4224c8572548342b7e64f091aba177da419/statuses" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/:ORG/:REPO/commits/2961d4224c8572548342b7e64f091aba177da419/statuses</a></p>
<pre><code class="lang-auto">[
  {
    "state" : "success",
    "creator" : {
      "id" : 272616,
      "organizations_url" : "https:\/\/api.github.com\/users\/:ORG\/orgs",
      "received_events_url" : "https:\/\/api.github.com\/users\/:ORG\/received_events",
      "following_url" : "https:\/\/api.github.com\/users\/:ORG\/following{\/other_user}",
      "login" : ":ORG",
      "avatar_url" : "https:\/\/avatars1.githubusercontent.com\/u\/272616?v=4",
      "url" : "https:\/\/api.github.com\/users\/:ORG",
      "node_id" : "MDQ6VXNlcjI3MjYxNg==",
      "subscriptions_url" : "https:\/\/api.github.com\/users\/:ORG\/subscriptions",
      "repos_url" : "https:\/\/api.github.com\/users\/:ORG\/repos",
      "type" : "User",
      "html_url" : "https:\/\/github.com\/:ORG",
      "events_url" : "https:\/\/api.github.com\/users\/:ORG\/events{\/privacy}",
      "site_admin" : false,
      "starred_url" : "https:\/\/api.github.com\/users\/:ORG\/starred{\/owner}{\/repo}",
      "gists_url" : "https:\/\/api.github.com\/users\/:ORG\/gists{\/gist_id}",
      "gravatar_id" : "",
      "followers_url" : "https:\/\/api.github.com\/users\/:ORG\/followers"
    },
    "id" : 10843188632,
    "created_at" : "2020-09-23T08:21:53Z",
    "context" : "codeclimate\/total-coverage",
    "target_url" : "https:\/\/codeclimate.com\/repos\/0000653a329424025b0003b0\/compare\/0501fe77fe21369a9cedde93ab8a04463081f325...2961d4224c8572548342b7e64f091aba177da419#ratings",
    "node_id" : "MDEzOlN0YXR1c0NvbnRleHQxMDg0MzE4ODYzMg==",
    "description" : "97% (0.0% change)",
    "updated_at" : "2020-09-23T08:21:53Z",
    "avatar_url" : "https:\/\/avatars3.githubusercontent.com\/oa\/26380?v=4",
    "url" : "https:\/\/api.github.com\/repos\/:ORG\/:REPO\/statuses\/2961d4224c8572548342b7e64f091aba177da419"
  },
  {
    "state" : "success",
    "creator" : {
      "id" : 272616,
      "organizations_url" : "https:\/\/api.github.com\/users\/:ORG\/orgs",
      "received_events_url" : "https:\/\/api.github.com\/users\/:ORG\/received_events",
      "following_url" : "https:\/\/api.github.com\/users\/:ORG\/following{\/other_user}",
      "login" : ":ORG",
      "avatar_url" : "https:\/\/avatars1.githubusercontent.com\/u\/272616?v=4",
      "url" : "https:\/\/api.github.com\/users\/:ORG",
      "node_id" : "MDQ6VXNlcjI3MjYxNg==",
      "subscriptions_url" : "https:\/\/api.github.com\/users\/:ORG\/subscriptions",
      "repos_url" : "https:\/\/api.github.com\/users\/:ORG\/repos",
      "type" : "User",
      "html_url" : "https:\/\/github.com\/:ORG",
      "events_url" : "https:\/\/api.github.com\/users\/:ORG\/events{\/privacy}",
      "site_admin" : false,
      "starred_url" : "https:\/\/api.github.com\/users\/:ORG\/starred{\/owner}{\/repo}",
      "gists_url" : "https:\/\/api.github.com\/users\/:ORG\/gists{\/gist_id}",
      "gravatar_id" : "",
      "followers_url" : "https:\/\/api.github.com\/users\/:ORG\/followers"
    },
    "id" : 10843188597,
    "created_at" : "2020-09-23T08:21:53Z",
    "context" : "codeclimate\/diff-coverage",
    "target_url" : "https:\/\/codeclimate.com\/repos\/0000653a329424025b0003b0\/compare\/0501fe77fe21369a9cedde93ab8a04463081f325...2961d4224c8572548342b7e64f091aba177da419#files-changed",
    "node_id" : "MDEzOlN0YXR1c0NvbnRleHQxMDg0MzE4ODU5Nw==",
    "description" : "100% (90% threshold)",
    "updated_at" : "2020-09-23T08:21:53Z",
    "avatar_url" : "https:\/\/avatars3.githubusercontent.com\/oa\/26380?v=4",
    "url" : "https:\/\/api.github.com\/repos\/:ORG\/:REPO\/statuses\/2961d4224c8572548342b7e64f091aba177da419"
  },
  {
    "state" : "success",
    "creator" : {
      "id" : 272616,
      "organizations_url" : "https:\/\/api.github.com\/users\/:ORG\/orgs",
      "received_events_url" : "https:\/\/api.github.com\/users\/:ORG\/received_events",
      "following_url" : "https:\/\/api.github.com\/users\/:ORG\/following{\/other_user}",
      "login" : ":ORG",
      "avatar_url" : "https:\/\/avatars1.githubusercontent.com\/u\/272616?v=4",
      "url" : "https:\/\/api.github.com\/users\/:ORG",
      "node_id" : "MDQ6VXNlcjI3MjYxNg==",
      "subscriptions_url" : "https:\/\/api.github.com\/users\/:ORG\/subscriptions",
      "repos_url" : "https:\/\/api.github.com\/users\/:ORG\/repos",
      "type" : "User",
      "html_url" : "https:\/\/github.com\/:ORG",
      "events_url" : "https:\/\/api.github.com\/users\/:ORG\/events{\/privacy}",
      "site_admin" : false,
      "starred_url" : "https:\/\/api.github.com\/users\/:ORG\/starred{\/owner}{\/repo}",
      "gists_url" : "https:\/\/api.github.com\/users\/:ORG\/gists{\/gist_id}",
      "gravatar_id" : "",
      "followers_url" : "https:\/\/api.github.com\/users\/:ORG\/followers"
    },
    "id" : 10843101579,
    "created_at" : "2020-09-23T08:13:21Z",
    "context" : "codeclimate",
    "target_url" : "https:\/\/codeclimate.com\/repos\/0000653a329424025b0003b0\/pull\/323",
    "node_id" : "MDEzOlN0YXR1c0NvbnRleHQxMDg0MzEwMTU3OQ==",
    "description" : "All good!",
    "updated_at" : "2020-09-23T08:13:21Z",
    "avatar_url" : "https:\/\/avatars3.githubusercontent.com\/oa\/26380?v=4",
    "url" : "https:\/\/api.github.com\/repos\/:ORG\/:REPO\/statuses\/2961d4224c8572548342b7e64f091aba177da419"
  },
  {
    "state" : "pending",
    "creator" : {
      "id" : 272616,
      "organizations_url" : "https:\/\/api.github.com\/users\/:ORG\/orgs",
      "received_events_url" : "https:\/\/api.github.com\/users\/:ORG\/received_events",
      "following_url" : "https:\/\/api.github.com\/users\/:ORG\/following{\/other_user}",
      "login" : ":ORG",
      "avatar_url" : "https:\/\/avatars1.githubusercontent.com\/u\/272616?v=4",
      "url" : "https:\/\/api.github.com\/users\/:ORG",
      "node_id" : "MDQ6VXNlcjI3MjYxNg==",
      "subscriptions_url" : "https:\/\/api.github.com\/users\/:ORG\/subscriptions",
      "repos_url" : "https:\/\/api.github.com\/users\/:ORG\/repos",
      "type" : "User",
      "html_url" : "https:\/\/github.com\/:ORG",
      "events_url" : "https:\/\/api.github.com\/users\/:ORG\/events{\/privacy}",
      "site_admin" : false,
      "starred_url" : "https:\/\/api.github.com\/users\/:ORG\/starred{\/owner}{\/repo}",
      "gists_url" : "https:\/\/api.github.com\/users\/:ORG\/gists{\/gist_id}",
      "gravatar_id" : "",
      "followers_url" : "https:\/\/api.github.com\/users\/:ORG\/followers"
    },
    "id" : 10843095531,
    "created_at" : "2020-09-23T08:12:40Z",
    "context" : "codeclimate",
    "target_url" : "https:\/\/codeclimate.com\/repos\/0000653a329424025b0003b0\/pull\/323",
    "node_id" : "MDEzOlN0YXR1c0NvbnRleHQxMDg0MzA5NTUzMQ==",
    "description" : "Code Climate is analyzing this code.",
    "updated_at" : "2020-09-23T08:12:40Z",
    "avatar_url" : "https:\/\/avatars3.githubusercontent.com\/oa\/26380?v=4",
    "url" : "https:\/\/api.github.com\/repos\/:ORG\/:REPO\/statuses\/2961d4224c8572548342b7e64f091aba177da419"
  }
]
</code></pre>
<p>Is this something expected?</p>
<p>Thanks</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/sigfrid">@sigfrid</a> ,</p>
<p>Please try using the “<a href="https://docs.github.com/en/rest/reference/checks#list-check-runs-for-a-git-reference" rel="noopener nofollow ugc"><strong>List check runs for a Git reference</strong></a>” API. It should work.</p>

**Comment by sigfrid** [comment]
<p>Hi <a class="mention" href="/brightran">@brightran</a><br>
The check runs API works.<br>
It’s a pity the GitHub Actions checks and third-party statuses don’t share the same API.</p>
<p>Have a good one,<br>
Sih</p>
